instrumentation/struts-2/src/main/java/io/opentelemetry/javaagent/instrumentation/struts2/Struts2Tracer.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.struts2;
+
+import com.opensymphony.xwork2.ActionInvocation;
+import com.opensymphony.xwork2.ActionProxy;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.servlet.ServletContextPath;
+import io.opentelemetry.instrumentation.api.tracer.BaseTracer;
+
+public class Struts2Tracer extends BaseTracer {
+
+  public static final Struts2Tracer TRACER = new Struts2Tracer();
+
+  public Span startSpan(ActionInvocation actionInvocation) {
+    Object action = actionInvocation.getAction();
+    Class<?> actionClass = action.getClass();
+
+    String method = actionInvocation.getProxy().getMethod();
+    String spanName = spanNameForMethod(actionClass, method);
+
+    Span strutsSpan = tracer.spanBuilder(spanName).startSpan();
+
+    strutsSpan.setAttribute("code.namespace", actionClass.getName());
+    if (method != null) {
+      strutsSpan.setAttribute("code.function", method);
```

Comment:
can you submit a PR to opentelemetry-java to add these to `SemanticAttributes`?

Reply:
Will do.

Reply:
Do I understand correctly, that to update `SemanticAttributes.java` I have to update https://github.com/open-telemetry/opentelemetry-specification/blob/master/semantic_conventions/trace/general.yaml from which `SemanticAttributes.java` s eventually generated?

Reply:
oh yes, good find, I believe that's the correct spot

Reply:
https://github.com/open-telemetry/opentelemetry-specification/pull/1229

Reply:
@trask are you Ok with merging this PR before we have spec updated, right?

Reply:
yes 👍